.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,16 @@
+name: Docker notebook tests
+
+on:
+  pull_request:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run the full docker profile
+        run: docker compose --profile full up -d
+      - name: Test notebooks in the docker environment
+        shell: bash
+        run: docker exec qs-jupyter "bash" "-c" "pip install nbmake pytest && pytest --nbmake /home/jovyan/serverless/"
```

Comment:
@psschwei any idea why the gateway doesnt seem to be working in the notebooks when I run these tests?

The gateway container should be up and available from line 13, I'm able to pip install the packages and run the nbmake command fine, the notebooks just fail bc they cant access gateway host

Reply:
I'm hitting a similar issue in reviewing #640 , still investigating